src/integrationTest/kotlin/com/sourcegraph/cody/chat/ChatTest.kt

```diff
@@ -0,0 +1,102 @@
+package com.sourcegraph.cody.chat
+
+import com.intellij.testFramework.runInEdtAndGet
+import com.intellij.testFramework.runInEdtAndWait
+import com.sourcegraph.cody.chat.ui.ContextFileActionLink
+import com.sourcegraph.cody.context.ui.EnterpriseEnhancedContextPanel
+import com.sourcegraph.cody.history.HistoryService
+import com.sourcegraph.cody.history.state.EnhancedContextState
+import com.sourcegraph.cody.history.state.RemoteRepositoryState
+import com.sourcegraph.cody.util.CodyIntegrationTextFixture
+import com.sourcegraph.cody.util.CustomJunitClassRunner
+import com.sourcegraph.cody.util.TestingCredentials
+import java.awt.Component
+import java.awt.Container
+import java.util.concurrent.TimeUnit
+import junit.framework.TestCase
+import org.awaitility.kotlin.await
+import org.awaitility.kotlin.until
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(CustomJunitClassRunner::class)
+class ChatTest : CodyIntegrationTextFixture() {
+  override fun myCredentials() = TestingCredentials.enterprise
+
+  @Test
+  fun testRemoteContextFileItems() {
+    val enhancedContextState =
+        EnhancedContextState().apply {
+          remoteRepositories.add(
+              RemoteRepositoryState().apply {
+                isEnabled = true
+                remoteUrl = "github.com/sourcegraph/cody"
+                codebaseName = "github.com/sourcegraph/cody"
+              })
+        }
+    HistoryService.getInstance(project).updateDefaultContextState(enhancedContextState)
+
+    val session = runInEdtAndGet { AgentChatSession.createNew(project) }
+
+    await.atMost(5, TimeUnit.SECONDS) until
+        {
+          (session.getPanel().contextView as EnterpriseEnhancedContextPanel)
+              .controller
+              .getConfiguredState()
+              .find { it.name == "github.com/sourcegraph/cody" } != null
+        }
+
+    runInEdtAndWait { session.sendMessage("How do we use JSON RPC in Cody?", emptyList()) }
+
+    await.atMost(15, TimeUnit.SECONDS) until
+        {
+          val messages = session.messages
+          messages.size == 2 &&
+              !messages[0].contextFiles.isNullOrEmpty() &&
+              !messages[1].text.isNullOrBlank()
+        }
```

Comment:
I don't like this style of assertions. 

If the condition doesn't hold, it will just throw a timeout exception, regardless of the failure mode. I would want to distinguish between an unexpected response (e.g. 3 messages instead of 2), and a complete lack of response.

Moreover, if I understand the [documentation](https://github.com/awaitility/awaitility/wiki/Usage) correctly, this code will wait for 15 seconds even if the response arrives immediately.

Reply:
> Moreover, if I understand the [documentation](https://github.com/awaitility/awaitility/wiki/Usage) correctly, this code will wait for 15 seconds even if the response arrives immediately.

This is not true. `atMost` waits at most the given time. Having the recordings ready the response arrives in <1s.

Reply:
Okay, but I still think it would be nice to distinguish between different failure modes.

Reply:
done